machine/v1alpha1/provider_types.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrlconfigv1 "sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	ProviderFinalizer         = "provider.cluster.x-k8s.io"
+	ConfigMapVersionLabelName = "provider.cluster.x-k8s.io/version"
+)
+
+// ProviderSpec is the desired state of the Provider.
+type ProviderSpec struct {
+	// Version indicates the provider version.
```

Comment:
What is an example of a provider and what sort of versions are there?

Reply:
An example is CAPI AWS providers and version represents specific release https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/tag/v1.3.0